src/Authenticator/AuthenticatorCollection.php

```diff
@@ -65,12 +65,17 @@ protected function _create($className, $alias, $config)
     /**
      * Resolves authenticator class name.
      *
-     * @param string $class Class name to be resolbed.
-     * @return string
+     * @param string $class Class name to be resolved.
+     * @return string|null
      */
     protected function _resolveClassName($class)
     {
-        return App::className($class, 'Authenticator', 'Authenticator');
+        $className = App::className($class, 'Authenticator', 'Authenticator') ?: null;
+        if (is_bool($class)) {
```

Comment:
This is equally bad as the casting imo, quite unnecesary :stuck_out_tongue:.  Stop trying to please phpstan :smile:.